multiqc/modules/supernova/supernova.py

```diff
@@ -0,0 +1,440 @@
+from collections import OrderedDict
+import logging
+import re
+import json
+from copy import deepcopy
+from multiqc import config
+from multiqc.plots import table, linegraph, bargraph
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+
+    def __init__(self):
+        super(MultiqcModule, self).__init__(name='Supernova', anchor='supernova',
+        href="https://www.10xgenomics.com/",
+        info="is a de novo genome assembler 10X Genomics linked-reads.")
+
+        # Table headers for the General data
+        self.gheaders = OrderedDict()
+
+
+        # Headers for the supernova Table
+        self.headers = OrderedDict()
+        self.headers['Asm size'] = {
+                'description': 'assembly size (in megabases) ;only scaffolds >= 10 kb',
+                'modify': lambda x: x / 1000000.0,
+                'suffix': 'Mb',
+                'scale': 'YlGn'
+        }
+        self.headers['# Long scaffs'] = {
+                'description': 'number of scaffolds >= 10 kb',
+                'scale': 'YlGn',
+                'format': '{:,.0f}',
+        }
+        self.headers['Scaff N50'] = {
+                'description': 'N50 scaffold size (in kilobases)',
+                'modify': lambda x: x / 1000.0,
+                'suffix': 'Kb',
+                'scale': 'RdYlGn'
+        }
+        self.headers['Phase N50'] = {
+                'description': 'N50 phase block size (in kilobases)',
+                'modify': lambda x: x / 1000.0,
+                'suffix': 'Kb',
+                'scale': 'RdYlGn',
+                'hidden': True
+        }
+        self.headers['Contig N50'] = {
+                'description': 'N50 contig size (in kilobases)',
+                'modify': lambda x: x / 1000.0,
+                'suffix': 'Kb',
+                'scale': 'RdYlGn',
+                'hidden': True
+        }
+        self.headers['Edge N50'] = {
+                'description': 'N50 edge size (in kilobases)',
+                'modify': lambda x: x / 1000.0,
+                'suffix': 'Kb',
+                'scale': 'RdYlGn',
+                'hidden': True
+        }
+        self.headers['Mol size'] = {
+                'description': 'weighted mean molecule size (in kilobases); ideal 50-100',
+                'modify': lambda x: x / 1000.0,
+                'suffix': 'Kb',
+                'scale': 'BuGn'
+        }
+        self.headers['Read len'] = {
+                'description': 'mean read length (in bases) after trimming; ideal 140',
+                'suffix': 'b',
+                'scale': 'PuBu',
+                'format': '{:,.0f}',
+                'hidden': True
+        }
+        self.headers['# Reads'] = {
+                'description': 'number of reads (in millions); ideal 800M-1200M for human',
+                'modify': lambda x: x / 1000000.0,
+                'suffix': 'M',
```

Comment:
Instead of having an `M` suffix, I prefer (and most other modules do this) to have the `M` in the column name...

```python
self.headers['# Reads'] = {
    'name': '# Reads (M)',
    'description': 'number of reads (in millions); ideal 800M-1200M for human',
    'modify': lambda x: x / 1000000.0,
    'scale': 'PuBu',
}
```

Reply:
I was merely trying to save precious horizontal screen space :smirk:
Or does giving a (longer) `name` option something special with the column widths?

Reply:
Hmm, `'name': '# Reads (M)'` didn't really do anything. The columns still said `# Reads`

Reply:
Sorry, it's `title` not `name`. I should really read my own documentation...

Reply:
And no, nothing special with column widths. This is just about consistency with other modules.

Reply:
No problem :joy:
Maybe do the same for Coverage? Change it to `Coverage (x)`. What about the (mega/kilo) bases units? Do you want me to move those units to the column title as well?

Reply:
Meh, it's probably fine..